portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFileEntryLocalServiceImpl.java

```diff
@@ -2795,8 +2800,10 @@ private void _expireFileEntriesByCompanyId(
 			}
 			else {
 				_expireFileVersion(
-					userId, fileEntry, latestFileVersion, true, workflowContext,
-					serviceContext);
+					userId, fileEntry, dlFileVersion,
+					dlFileVersion.getFileVersionId() ==
+					latestFileVersion.getFileVersionId(),
```

Comment:
(optional) Not sure about this expression. Brian may prefer a variable + explicit conditional assignment.

Reply:
Thanks for your suggestion

I do not like it because it generates more code
```
	if (expireAllVersions.isEnabled()) {
		List<DLFileVersion> fileVersions = fileEntry.getFileVersions(
			WorkflowConstants.STATUS_ANY);

		for (DLFileVersion fileVersion : fileVersions) {
			boolean lastFileVersion = false;

			if (fileVersion.getFileVersionId() ==
					latestFileVersion.getFileVersionId()) {

				lastFileVersion = true;
			}

			_expireFileVersion(
				userId, fileEntry, fileVersion, lastFileVersion,
				workflowContext, serviceContext);
		}
	}
	else {
		boolean lastFileVersion = false;

		if (dlFileVersion.getFileVersionId() ==
				latestFileVersion.getFileVersionId()) {

			lastFileVersion = true;
		}

		_expireFileVersion(
			userId, fileEntry, dlFileVersion, lastFileVersion,
			workflowContext, serviceContext);
	}
```

but I trust your intuition and your expertise.

I added a commit extracting the variables.

Thanks

Reply:
ok, method created, to make things more readable